cvmfs/upload_gateway.cc

```diff
@@ -231,6 +231,10 @@ bool GatewayUploader::ReadKey(const std::string& key_file, std::string* key_id,
   return gateway::ReadKeys(key_file, key_id, secret);
 }
 
+int64_t GatewayUploader::DoGetObjectSize(const std::string &file_name) {
+  return 0;
```

Comment:
I think we should return `-EOPNOTSUPP` (or another suitable error code from errno.h) in order to distinguish it from a zero-byte object.